filebeat/prospector/prospector.go

```diff
@@ -31,18 +33,20 @@ type Prospector struct {
 	wg               sync.WaitGroup
 	channelWg        sync.WaitGroup // Separate waitgroup for channels as not stopped on completion
 	harvesterCounter uint64
+	Hash             uint64
```

Comment:
how about `s/Hash uint64/ID ProspectorID` with `ProspectorID string`? e.g. encode the prospector type as well: `<prospector-type>-<config-hash>`. if we ever introduce named prospectors the string ca be easily adapted to `<prospector-type>-<name>-<config-hash>`.